docs/Practices/hub/hub.md

```diff
@@ -0,0 +1,118 @@
+
+## Guidelines
+- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Coupling and Cohesion](./learning.md#coupling-and-cohesion) Other than the consumption of a command or query, code sharing between bounded contexts should be avoided by default. There may be cases where a bounded context is implemented in a different project. In this case no dependency should be had on the code of that project and integration must be done asynchronously (http call, event handler, etc).
+
+- [UA: Vertical Slices](./learning.md#vertical-slices), [UA: Coupling and Cohesion](./learning.md#coupling-and-cohesion) The only dependencies between Vertical Slices (different command and queries) should be usages of the command or query function from another slice. Using shared code that sits outside the slice is fine. There are occasional exceptions to this rule, but it is preferred to err on the side of duplication by default. If you absolutely need to share code between slices, it should exist in a `shared` directory at the bounded context root. *Vertical Slices revolve around the axis of change being a slice. We want to couple along the axis of change. Sometimes we might be tempted to share code between two slices. One example would be needing to share some type, because two slices use a similar shape. An issue arises though when one slice is iterated on and needs to change that shape while another slice doesn't. For this reason it would have been better to duplicate the shape between slices. A rule of thumb is if you plan to share code (type, function, constant, etc.), you need to be VERY sure it represents the exact same concept between slices. Sometimes shared code might actually fit best in an existing shared layer (repository, domain, data)*
+
+- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Clean Architecture](./learning.md#cleanhexagonalonion-architecture), [UA: Derived Types](./learning.md#derived-types), [UA: Coupling and Cohesion](./learning.md#coupling-and-cohesion) Base types representing the shape of domain models should be in the domain directory at the bounded context root. These types can be shared, often for the purpose of derivatives, among slices.
+
+- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Clean Architecture](./learning.md#cleanhexagonalonion-architecture), [UA: Coupling and Cohesion](./learning.md#coupling-and-cohesion) Domain types, regardless of whether scoped to a slice or scoped to a bounded context, should not depend on code that is not itself either domain code or is a generic utility type (error handling, ts utils, and such). *This is a concept borrowed from Clean/Hexagonal/Onion architecture and key piece in the internal consistency of our domain model if followed. Exceptions to this rule should be very slim or non existent.*
```

Comment:
little repeat here.

Reply:
removed explainer text to reduce repeat.